src/cpp/session/modules/clang/CodeCompletion.cpp

```diff
@@ -154,6 +163,244 @@ core::json::Object toJson(const CodeCompleteResult& result)
    return resultJson;
 }
 
+void discoverTranslationUnitIncludePaths(const FilePath& filePath,
+                                         std::vector<std::string>* pIncludePaths)
+{
+   std::vector<std::string> args =
+         rCompilationDatabase().compileArgsForTranslationUnit(
+            filePath.absolutePathNative(), false);
+   
+   BOOST_FOREACH(const std::string& arg, args)
+   {
+      if (boost::algorithm::starts_with(arg, "-I"))
+      {
+         // skip RStudio's libclang builtin header paths
+         if (arg.find("libclang/builtin-headers") != std::string::npos)
+            continue;
+         
+         pIncludePaths->push_back(
+                  string_utils::strippedOfQuotes(arg.substr(2)));
+      }
+   }
+}
+
+void discoverSystemIncludePaths(std::vector<std::string>* pIncludePaths)
+{
+   core::system::ProcessOptions processOptions;
+   processOptions.redirectStdErrToStdOut = true;
+   
+   // resolve R CMD location for shell command
+   FilePath rBinDir;
+   Error error = module_context::rBinDir(&rBinDir);
+   if (error)
+   {
+      LOG_ERROR(error);
+      return;
+   }
+   
+   shell_utils::ShellCommand rCmd = module_context::rCmd(rBinDir);
+   
+   // get the CXX compiler by asking R
+   std::string compilerPath;
```

Comment:
Possibly. Let me get through the next couple of days (the release) then
will once again take a closer look at this PR.

On Sun, Feb 7, 2016 at 3:16 PM, Kevin Ushey <notifications@github.com>
wrote:

> In src/cpp/session/modules/clang/CodeCompletion.cpp
> <https://github.com/rstudio/rstudio/pull/578#discussion_r52124011>:
>
> > +   core::system::ProcessOptions processOptions;
> > +   processOptions.redirectStdErrToStdOut = true;
> > +
> > +   // resolve R CMD location for shell command
> > +   FilePath rBinDir;
> > +   Error error = module_context::rBinDir(&rBinDir);
> > +   if (error)
> > +   {
> > +      LOG_ERROR(error);
> > +      return;
> > +   }
> > +
> > +   shell_utils::ShellCommand rCmd = module_context::rCmd(rBinDir);
> > +
> > +   // get the CXX compiler by asking R
> > +   std::string compilerPath;
>
> Could we also get this using the R compilation database?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rstudio/rstudio/pull/578/files#r52124011>.
>